### ISSUE TITLE
`nix copy --all` from local binary cache fails

### ISSUE THREAD

**Issue by Kha** [NixOS/nix#6730]
**Describe the bug**

Trying to copy all paths from a local binary cache fails with `error: not an absolute path: 'nix-archive-1'`

**Steps To Reproduce**

```
$ nix copy nixpkgs#hello --to file:///tmp/temp-store
$ nix copy --all --from file:///tmp/temp-store
error: not an absolute path: 'nix-archive-1'
```

**Expected behavior**

It should succeed.

**`nix-env --version` output**

```
nix-env (Nix) 2.9.1
```

**Additional context**

The same error message, but in a different context and apparently not always reproducible, was logged here: https://github.com/NixOS/nix/issues/6253

**Comment by mupdt** [comment]
We are able to reproduce this bug reliably with nix 2.5.1.

While looking through the code, we found that #6621 could fix this issue. 

We successfully verified (with @thufschmitt's help) that #6621 does indeed fix this issue.

**Comment by Kha** [comment]
Did you link to the correct PR :thinking: ?

I mean, I wouldn't complain if I somehow fixed my own reported bug without noticing, but that one's about a pretty different topic and I can still reproduce the issue using it.

**Comment by thufschmitt** [comment]
I think this has been fixed by #6612 (except in that `ssh-ng` case which is tracked by #7720 ), so I'll close this. Feel free to reopen if it's not the case
